application/src/main/java/org/togetherjava/tjbot/features/github/GitHubLinkCommand.java

```diff
@@ -0,0 +1,119 @@
+
+    private static final String REPOSITORY_OWNER_OPTION = "owner";
+    private static final String REPOSITORY_NAME_OPTION = "name";
+
+    private final Database database;
+    private final String githubApiKey;
+
+    /**
+     * Creates new GitHub link command.
+     * 
+     * @param database the database to store the new linked pull request notifications
+     * @param config the config to get the GitHub API key
+     */
+    public GitHubLinkCommand(Database database, Config config) {
+        super("link-gh-project",
+                "Links a GitHub repository to this project post to receive pull request notifications",
+                CommandVisibility.GUILD);
+        this.database = database;
+        this.githubApiKey = config.getGitHubApiKey();
+
+        getData()
+            .addOption(OptionType.STRING, REPOSITORY_OWNER_OPTION,
+                    "The owner of the repository to be linked", true)
+            .addOption(OptionType.STRING, REPOSITORY_NAME_OPTION,
+                    "The name of the repository to be linked", true);
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        OptionMapping repositoryOwnerOption = event.getOption(REPOSITORY_OWNER_OPTION);
+        OptionMapping repositoryNameOption = event.getOption(REPOSITORY_NAME_OPTION);
+
+        if (repositoryOwnerOption == null || repositoryNameOption == null) {
+            event.reply("You must specify a repository owner and a repository name")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        long channelId = event.getChannelIdLong();
+        String repositoryOwner = repositoryOwnerOption.getAsString();
+        String repositoryName = repositoryNameOption.getAsString();
+
+        GitHub github;
+        try {
+            github = new GitHubBuilder().withOAuthToken(githubApiKey).build();
+        } catch (IOException e) {
+            logger.error("Failed to initialize GitHub API wrapper.", e);
+            event.reply("Failed to initialize GitHub API wrapper.").setEphemeral(true).queue();
+            return;
+        }
+
+        try {
+            if (!isRepositoryAccessible(github, repositoryOwner, repositoryName)) {
+                event.reply("Repository is not publicly available.").setEphemeral(true).queue();
+                logger.info("Repository {}/{} is not accessible.", repositoryOwner, repositoryName);
+                return;
+            }
+        } catch (IOException e) {
+            logger.error("Failed to check if GitHub repository is available.", e);
+            event.reply("Failed to link repository.").setEphemeral(true).queue();
+            return;
+        }
+
+        try {
+            saveNotificationToDatabase(channelId, repositoryOwner, repositoryName);
+            event.reply("Successfully linked repository.").setEphemeral(true).queue();
```

Comment:
Let's make this response a pretty embed for everyone in the project thread to see.